### PR TITLE
blueprints/instance-initializers-test: Add RFC232 variants

### PR DESCRIPTION
**Pull request by snewcomer**
Progress on #15933

### CODE REVIEW

File: blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      this.application.deferReadiness();
+    });
+  });
+  hooks.afterEach(function() {
+    destroyApp(this.application);
+  });
+
+  // Replace this with your real tests.
+  test('it works', function(assert) {
+    initialize(this.application);
```
Comment (rwjblue) [inline blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: Instance initializers expect to receive an Ember.ApplicationInstance instance, but this is an Ember.Application instance.

File: blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import destroyApp from '../../helpers/destroy-app';
+
+module('<%= friendlyTestName %>', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function() {
+    run(() => {
+      this.application = Application.create();
```
Comment (rwjblue) [inline blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: Can you swap this to `Application.create({ autoboot: false })` here?

File: blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      this.application.deferReadiness();
```
Comment (rwjblue) [inline blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: This shouldn’t be needed with my suggested changes

File: blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import { initialize } from '<%= dasherizedModulePrefix %>/initializers/<%= dasherizedModuleName %>';
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import destroyApp from '../../helpers/destroy-app';
+
+module('<%= friendlyTestName %>', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function() {
+    run(() => {
```
Comment (rwjblue) [inline blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: We shouldn’t need to `Ember.run` here, why is this required? Is there an error or something without it?

File: blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js (hunk 1 above)
Comment (rwjblue) [reply blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: To fix that, you would do something like:

```
module(‘asdf’, function(hooks) {

  hooks.beforeEach(function() {
    this.Application = Ember.Application.extend();
    this.application = this.Application.create({ autoboot: true });
    this.application.instanceInitializer({
      name: ‘initializer under test’,
      initialize
    });
  });

  hooks.afterEach(function() {
    runDestroy(this.instance);
    runDestroy(this.application);
  });

  test(‘it works’, async function(assert) {
    this.instance = this.application.buildInstance();
    await this.instance.boot();
    assert.ok(didTheInitializerWork);
  });
});
```

**Review by rwjblue — CHANGES_REQUESTED** [review]
I’d like to restructure things a bit, I left an example in line would you mind giving that snippet a whirl and confirming that it works properly? I believe it should (barring any typos etc)

File: blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js (hunk 1 above)
Comment (snewcomer) [reply blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: @rwjblue Ah yeah overall my bad.  So with this test, `await this.instance.boot();` results in `Assertion Failed: You cannot use the same root element (BODY) multiple times in an Ember.Application`.  Failing on this line - https://github.com/emberjs/ember.js/blob/5f4ac602813d8fbd78f26fc955acf693e0b3d62a/packages/ember-views/lib/system/event_dispatcher.js#L179

File: blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js (hunk 4 above)
Comment (snewcomer) [reply blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: Yeah the error I get is the `You have turned on testing mode...` error.

File: blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js (hunk 1 above)
Comment (snewcomer) [reply blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: tested on a 2.17 application.

Comment (rwjblue) [reply blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: Would you mind pushing up your demo app so I can tinker with it to find the right combo?

Comment (rwjblue) [reply blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: Also, I made a really bad mistake in my comment above, it should be `{ autoboot: false }` (and I had `true` in the comment)

Comment (rwjblue) [reply blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js]: OK, just tested locally, and I think that it will work once updating to `{ autoboot: false }` (sorry about that).

* [example instance-initializer test](https://github.com/rwjblue/__emberjs-PR15945_demo/blob/master/tests/unit/instance-initializers/do-stuff-test.js)
* [diff of changes needed](https://github.com/rwjblue/__emberjs-PR15945_demo/commit/30a36e61691bf4c13ff0321726dc927b9aa93686)

**Comment by rwjblue** [comment]
Awesome, thank you!
